app/src/main/java/org/groundplatform/android/usecases/survey/ListAvailableSurveysUseCase.kt

```diff
@@ -57,8 +57,17 @@ constructor(
   private suspend fun getRemoteSurveyList(): Flow<List<SurveyListItem>> {
     val user = userRepository.getAuthenticatedUser()
     val remoteSurveyListFlow = remoteDataStore.getSurveyList(user)
-    return remoteSurveyListFlow.combine(getLocalSurveyList()) { remoteSurveys, localSurveys ->
-      remoteSurveys.map { remoteSurvey -> addOfflineStatus(remoteSurvey, localSurveys) }
+    val remotePublicSurveyListFlow = remoteDataStore.getPublicSurveyList()
+    val localSurveyListFlow = getLocalSurveyList()
+
+    return combine(remoteSurveyListFlow, remotePublicSurveyListFlow, localSurveyListFlow) {
+      remoteSurveys,
+      remotePublicSurveys,
+      localSurveys ->
+      val allRemoteSurveys = remoteSurveys + remotePublicSurveys
+      allRemoteSurveys
+        .distinctBy { it.id }
```

Comment:
You can already guarantee they're unique by adding a filter for RESTRICTED_VALUE on the query which builds `remoteSurveys`, just be sure to rename to `restrictedRemoteSurveys`.

Reply:
done